crates/assists/src/handlers/extract_module_to_file.rs

```diff
@@ -0,0 +1,94 @@
+use ide_db::source_change::FileSystemEdit;
+use vfs::FileId;
+use syntax::{
+    ast::{
+        self,
+        NameOwner
+    },
+    AstNode,
+    TextSize
+};
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+// Assist: extract_module_to_file
+//
+// This assist extract module to file.
+//
+// ```
+// mod tests {<|>
+//     #[test] fn t() {}
+// }
+// ```
+// ->
+// ```
+// mod tests;
+//
+// #[test] fn t() {}
+// ```
+pub(crate) fn extract_module_to_file(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let assist_id = AssistId("extract_module_to_file", AssistKind::RefactorExtract);
+    let assist_label = "Extract module to file";
+    // TODO: how to get krate_name?
+    let krate_name = "foo";
+    let module = ctx.find_node_at_offset::<ast::Module>()?;
+    let module_items = module.item_list()?;
+    let module_items_text = module_items.to_string();
+    let module_name = module.name()?;
+    let target = module.syntax().text_range();
+    // TODO: how to increment FileId for new module file?
+    let new_file_id = FileId(1);
```

Comment:
Ah, this is confusing... let me lend a refactor/doc improvment here to explain how to do this better...

Reply:
Here, #6784 should hopefully clarify the semantics of `anchor` field. In this particular case, you'll need to set it to the current file id (`AssistContext.frange.file_id`).